geoips/interfaces/module_based/output_checkers.py

```diff
@@ -0,0 +1,500 @@
+
+from geoips.interfaces.base import BaseModuleInterface, BaseModulePlugin
+import logging
+from geoips.errors import PluginError
+
+LOG = logging.getLogger(__name__)
+
+
+class OutputCheckersBasePlugin(BaseModulePlugin):
+    """Output Checkers Base Plugin for comparing data outputs."""
+
+    def get_out_diff_fname(self, compare_product, output_product, ext=None, flag=None):
+        """Obtain the filename for output and comparison product diff.
+
+        Parameters
+        ----------
+        compare_product : str
+            Full path to product filename in the comparison directory
+        output_product : str
+            Full path to product filename in the current output directory
+        ext : str, default=None
+            Extension to use as an alternative to the original file extension
+        flag : str, default=None
+            Additional identifying string to include in output diff filename
+
+        Returns
+        -------
+        out_diff_fname : str
+            Full path to output diff file.
+        """
+        from os import makedirs, getenv
+        from os.path import join, dirname, basename, exists, splitext
+
+        if not flag:
+            flag = ""
+        diffdir = join(
+            dirname(compare_product), "diff_test_output_dir_{0}".format(getenv("USER"))
+        )
+        out_diff_fname = join(
+            diffdir, "diff_test_output_" + flag + basename(output_product)
+        )
+        if not exists(diffdir):
+            makedirs(diffdir)
+        # Output jifs as tif for easy viewing.
+        if ext is not None:
+            out_diff_fname = splitext(out_diff_fname)[0] + ext
+        elif splitext(out_diff_fname)[-1] == ".jif":
+            out_diff_fname = splitext(out_diff_fname)[0] + ".png"
+        return out_diff_fname
+
+    def compare_outputs(self, compare_path, output_products, test_product_func=None):
+        """Compare the "correct" imagery found the list of current output_products.
+
+        Compares files produced in the current processing run with the list of
+        "correct" files contained in "compare_path".
+
+        Parameters
+        ----------
+        compare_path : str
+            Path to directory of "correct" products - filenames must match
+            output_products
+        output_products : list of str
+            List of strings of current output products,
+            to compare with products in compare_path
+        test_product_func : function, default=None
+            Alternative function to be used for testing output product
+
+            * Call signature must be:
+
+                * output_product, compare_product, goodcomps, badcomps, compare_strings
+
+            * Return must be:
+
+                * goodcomps, badcomps, compare_strings
+
+            * If None, use geoips.compare_outputs.test_product)
+
+        Returns
+        -------
+        int
+            Binary code: 0 if all comparisons were completed successfully.
+        """
```

Comment:
@evrose54 Maybe right at the beginning of the "compare_outputs" module just raise an informative error if "outputs_match" or "correct_type" are not defined?  Note this will likely not work for the gzip and gunzip ones (since they are slightly different...), but I *think* it is ok to just handle those explicitly and slightly differently (since they apply to any file format - it's just a way to gunzip before comparing what the type really is, and gzip when you're done).